content/200-concepts/100-components/02-prisma-client/250-tracing.mdx

```diff
@@ -0,0 +1,324 @@
+
+For example, given the following code:
+
+```ts
+await prisma.$transaction([
+  prisma.user.create({
+    data: {
+      email: email,
+    },
+  }),
+  prisma.user.findMany({
+    where: {
+      email: email,
+    },
+  }),
+])
+```
+
+The trace is structured as follows:
+
+- `prisma:transaction`
+  - `prisma`
+    - `prisma:itx_runner`
+      - `prisma:connection`
+      - `prisma:itx_query_builder`
+        - `prisma:db_query`
+      - `prisma:itx_query_builder`
+        - `prisma:db_query`
+        - `prisma:db_query`
+
+### Adding More Instrumentation
+
+A nice benefit of OpenTelemetry is the ability to add more instrumentation while making minimal changes to your application code.
+
+You can add HTTP and [ExpressJS](https://expressjs.com/) tracing by adding more instrumentations to your OpenTelemetry configuration. This instrumentation will add spans for the full request-response lifecycle showing you how long your HTTP requests take.
+
+```js
+// Imports
+import { ExpressInstrumentation } from '@opentelemetry/instrumentation-express'
+import { HttpInstrumentation } from '@opentelemetry/instrumentation-http'
+
+// Register your auto-instrumentors
+registerInstrumentations({
+  tracerProvider: provider,
+  instrumentations: [
+    new HttpInstrumentation(),
+    new ExpressInstrumentation(),
+    new PrismaInstrumentation(),
+  ],
+})
+```
+
+For a full list of available instrumentation, take a look at the [OpenTelemetry Registry](https://opentelemetry.io/registry/?language=js&component=instrumentation).
+
+### Customizing Resource Attributes
+
+You can adjust how your application's traces are grouped by changing the resource attributes to be more specific to your application:
+
+```js
+const provider = new NodeTracerProvider({
+  resource: new Resource({
+    [SemanticResourceAttributes.SERVICE_NAME]: 'weblog',
+    [SemanticResourceAttributes.SERVICE_VERSION]: '1.0.0',
+  }),
+})
+```
+
+There is an ongoing effort to standardize common resource attributes. Whenever possible, it's a good idea to follow the [standard attribute names](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/README.md).
+
+### Reducing performance impact
+
+If your application sends a large number of spans to a collector, this can have a significant performance impact. You can use the following approaches to reduce this impact:
+
+- [Use the BatchSpanProcessor](#send-traces-in-batches-using-the-batchspanprocessor)
+- [Send fewer spans to the collector](#send-fewer-spans-to-the-collector-via-sampling)
+
+#### Send Traces in Batches using the `BatchSpanProcessor`
+
+In a production environment, you can use the OpenTelemetry `BatchSpanProcessor` to send the spans to a collector in batches rather than one at a time. However, during development and testing, you might not want to send spans in batches. In this situation, you might prefer to use the `SimpleSpanProcessor`.
```

Comment:
```suggestion
In a production environment, you can use OpenTelemetry's `BatchSpanProcessor` to send the spans to a collector in batches rather than one at a time. However, during development and testing, you might not want to send spans in batches. In this situation, you might prefer to use the `SimpleSpanProcessor`.
```